picard/ui/options/interface.py

```diff
@@ -113,4 +199,130 @@ def starting_directory_browse(self):
             path = os.path.normpath(unicode(path))
             item.setText(path)
 
+    def _valid_action(self, name):
+        return name in self.TOOLBAR_BUTTONS or name == 'separator'
+
+    def _get_icon_from_name(self, name):
+        return self.TOOLBAR_BUTTONS[name]['icon']
+
+    def _insert_item(self, action, index=None):
+        list_item = ToolbarListItem(action)
+        list_item.setToolTip(_(u'Drag and Drop to re-order'))
+        if action in self.TOOLBAR_BUTTONS:
+            list_item.setText(_(self.TOOLBAR_BUTTONS[action]['label']))
+            list_item.setIcon(icontheme.lookup(self._get_icon_from_name(action), icontheme.ICON_SIZE_MENU))
+        else:
+            list_item.setText(self.SEPARATOR)
+        if index:
```

Comment:
Test here should be `index is not None`, because `index` may be zero (or not, but safer anyway).